src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -138,11 +139,32 @@ private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
         {
             if (!string.IsNullOrEmpty(loginHint))
             {
-                IReadOnlyList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
+                IList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
                     .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
                            a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
                     .ToList();
 
+                if (accounts.Count == 0 && DesktopOsHelper.IsWindows())

```

Comment:
Interesting idea. But what is the point here? To get AcquireTokenSilent + WAM to succeed when login_hint matches UPN of the Windows user?

I think this logic would need to move to the broker

Reply:
Can you please open a bug or a feature request to tie this PR to ?